src/deploy/hosting/prepare.ts

```diff
@@ -44,6 +44,22 @@ export async function prepare(context: any, options: any): Promise<void> {
       labels: deploymentTool.labels(),
     };
 
+    // Keep a copy of the data in the payload, to patch in later
+    payload.hosting ||= {};
+    payload.hosting.config ||= {};
+    payload.hosting.config[deploy.site] = { ...data.config };
+
+    // Filter out any rewrites that point to GCFv2 functions that are being deployed
+    Object.entries(payload.functions?.backend?.endpoints || {}).forEach(([region, endpoints]) => {
+      Object.entries(endpoints as {}).forEach(([serviceId, endpoint]) => {
+        if ((endpoint as any).platform === "gcfv2") {
+          data.config.rewrites = data.config.rewrites.filter(
+            (rewrite: any) => rewrite.run?.serviceId !== serviceId && rewrite.run?.region !== region
```

Comment:
The typing inside this loops seems a bit code smelly to me - I don't like that we are casting to {} then to any. I'd prefer to give this some strong types so someone unfamiliar with the code can easily know what to expect here.

Reply:
Yeah, just fighting Object.entries and any, strong types will help a lot.